mautrix_telegram/abstract_user.py

```diff
@@ -306,7 +306,16 @@ async def has_full_access(self, allow_bot: bool = False) -> bool:
     async def start(self, delete_unless_authenticated: bool = False) -> AbstractUser:
         if not self.client:
             await self._init_client()
-        await self.client.connect()
+        while True:
+            attempts = 1
+            try:
+                await self.client.connect()
+            except Exception:
+                self.log.exception("Exception connecting to Telegram, retrying in 5s...")
+                attempts += 1
+                if attempts > 10:
+                    raise
```

Comment:
Probably only log when not raising

```suggestion
                attempts += 1
                if attempts > 10:
                    raise
                self.log.exception("Exception connecting to Telegram, retrying in 5s...")
```